### ISSUE TITLE
Sometimes run well but sometimes raised an error while submiting a same command

### ISSUE THREAD

**Issue by zhangshd** [pharmai/plip#80]
**Describe the bug**
Dear developers,
I got an error when I extracted the interaction from `4gql`. But it works well sometimes. Is there something wrong with the reproducibility of PLIP?
```
Traceback (most recent call last):
  File "/src/plip/plipcmd.py", line 311, in <module>
    main()
  File "/src/plip/plipcmd.py", line 307, in main
    run_analysis(expanded_path, arguments.pdbid)  # Start main script
  File "/src/plip/plipcmd.py", line 160, in run_analysis
    process_pdb(pdbpath, config.OUTPATH, outputprefix=output_prefix)
  File "/src/plip/plipcmd.py", line 53, in process_pdb
    mol.load_pdb(pdbfile, as_string=as_string)
  File "/src/plip/structure/preparation.py", line 1367, in load_pdb
    ligandfinder = LigandFinder(self.protcomplex, self.altconf, self.modres, self.covalent, self.Mapper)
  File "/src/plip/structure/preparation.py", line 214, in __init__
    self.ligands = self.getligs()
  File "/src/plip/structure/preparation.py", line 259, in getligs
    ligands.append(self.extract_ligand(kmer))
  File "/src/plip/structure/preparation.py", line 302, in extract_ligand
    del cur_hetatoms[atom_id]
KeyError: 2708
```

**To Reproduce**
Repeating the command  `plip -i 4gql -yt`  several times

**Expected behavior**
This bug be repaired.

**Screenshots**
![](https://s1.ax1x.com/2020/08/22/daktoQ.png).

**Desktop (please complete the following information):**
 - OS: window 10 + docker
- plip: 2.1.4

Thank you for your support!

Best,
Shengde

**Comment by fkaiserbio** [comment]
Dear Shengde,

thank you for your detailed report. In order to have a deterministic look on it, could you please provide the `*_protonated.pdb` file written by PLIP of a run that fails?

Thanks
Florian

**Comment by zhangshd** [comment]
> Dear Shengde,
> 
> thank you for your detailed report. In order to have a deterministic look on it, could you please provide the `*_protonated.pdb` file written by PLIP of a run that fails?
> 
> Thanks
> Florian

Dear Florian,

thank you for you help.  I have to say that PLIP  didn't write any thing except the downloaded `4gql.pdb` file in the failed examples. I can reproduce this failure whether in my local environment or in docker by using the command `plip -i 4gql -yt`.  I suppose that you may reproduce this failure too. If yes, the reproduced failure can be helpful.

Best,
Shengde

**Comment by fkaiserbio** [comment]
Dear Shengde,

okay, that's interesting and certainly points to a bug. When I find time I will look into this, but it might take some time. Thanks for reporting!

Best
Florian

**Comment by fkaiserbio** [comment]
Dear Shengde,

sorry for the delay. The issue is fixed in 84f91226a27fe209438f5b64ca8c1a75ff526d50 and will be merged in the next release. The issue was that (for the rare case) where a composite ligand contains alternative locations, these were not correctly indexed before removal.
